src/Application/Startup/GameStarter.cpp

```diff
@@ -58,82 +58,54 @@
 constexpr std::string_view configName = "openenroth.ini";
 
 GameStarter::GameStarter(GameStarterOptions options): _options(std::move(options)) {
-    // Init environment.
-    _environment = Environment::createStandardEnvironment();
-
-    // Init default log sink.
-    _rootLogSink = std::make_unique<DistLogSink>();
-    _defaultLogSink = LogSink::createDefaultSink();
-    _rootLogSink->addLogSink(_defaultLogSink.get());
-
-    if (_options.logLevel) {
-        // If we know the log level - init logger properly right away.
-        _logger = std::make_unique<Logger>(*_options.logLevel, _rootLogSink.get());
-    } else {
-        // Otherwise log into a buffer before we read the log level from config.
-        _bufferLogSink = std::make_unique<BufferLogSink>();
-        _logger = std::make_unique<Logger>(LOG_TRACE, _bufferLogSink.get());
-    }
+    // Init logging.
+    _logStarter.initPrimary();
+    Engine::LogEngineBuildInfo();
 
     try {
-        initializeWithLogger();
+        initWithLogger();
     } catch (const std::exception &e) {
-        // If we still haven't figured out the log level - print everything out at LOG_TRACE.
-        if (_bufferLogSink) {
-            _logger->setLevel(LOG_TRACE);
-            _logger->setSink(_rootLogSink.get());
-            _bufferLogSink->flush(_logger.get());
-        }
-
-        // Also log the exception so that it goes to all registered loggers.
-        _logger->critical("Terminated with exception: {}", e.what());
-
-        // And propagate.
+        logger->critical("Terminated with exception: {}", e.what());
         throw;
     }
 }
 
-void GameStarter::initializeWithLogger() {
-    // Log engine info.
-    Engine::LogEngineBuildInfo();
+void GameStarter::initWithLogger() {
+    // Init environment.
+    _environment = Environment::createStandardEnvironment();
 
-    // Init paths.
-    resolvePaths(_environment.get(), &_options, _logger.get());
+    // Resolve user path, create user fs & file logger.
+    resolveUserPath(_environment.get(), &_options);
+    _fsStarter.initUserFs(_options.ramFsUserData, _options.userPath);
+    _logStarter.initSecondary(ufs);
 
-    // Init filesystem - needs data paths initialized.
-    _fs = std::make_unique<EngineFileSystem>(_options.dataPath, _options.userPath);
-    if (_options.ramFsUserData) {
-        _userRamFs = std::make_unique<MemoryFileSystem>("ramfs");
-        ufs = _userRamFs.get();
-    }
+    // Resolve data path, create data fs.
+    // TODO(captainurist): actually move datapath to config?
```

Comment:
Either that or have it as a fall back option?

Reply:
Yeah, if I add an option in config it'll work as a fallback, will still be able to override the path via command line. I'd rather drop the env vars in this case though.